crates/consensus/src/engine/mod.rs

```diff
@@ -291,3 +321,148 @@ where
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use assert_matches::assert_matches;
+    use reth_interfaces::test_utils::generators::random_block;
+    use reth_provider::test_utils::MockEthProvider;
+    use tokio::sync::mpsc::unbounded_channel;
+
+    // https://github.com/ethereum/execution-apis/blob/main/src/engine/paris.md#specification-3
+    mod exchange_transition_configuration {
+        use super::*;
+
+        #[tokio::test]
+        async fn terminal_td_mismatch() {
+            let (tx, rx) = unbounded_channel();
+            let config = Config::default();
+            let engine = EthConsensusEngine {
+                client: Arc::new(MockEthProvider::default()),
+                config: config.clone(),
+                local_store: Default::default(),
+                rx: UnboundedReceiverStream::new(rx),
```

Comment:
Should this be unbounded? @mattsse

Reply:
`EthConsensusEngine` is just an endless future. Hence the unbounded stream for handling requests